packages/provider/src/transports/base-provider-transport.ts

```diff
@@ -345,9 +357,27 @@ export abstract class BaseProviderTransport implements ProviderTransport {
       })
     })
 
-    return Promise.race<WalletSession>([
-      connect,
-      closeWallet
+    return Promise.race<WalletSession>([connect, closeWallet])
+  }
+
+  waitUntilAuthorized = async (): Promise<ConnectDetails> => {
```

Comment:
lets remove and just have waitUntilConnected(). If a dapp makes a request to `connect({ requestAuthorization })`, then we need to respond with connect and inside provide the authorization